src/main/java/com/example/music/backend/verification/RegistrationListener.java

```diff
@@ -0,0 +1,54 @@
+package com.example.music.backend.verification;
+
+import com.example.music.backend.user.domain.User;
+import com.example.music.backend.user.service.UserService;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.MessageSource;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class RegistrationListener implements
+        ApplicationListener<OnRegistrationCompleteEvent> {
+
+
+    private UserService userService;
+
+
+    private MessageSource messages;
+
+
+    private JavaMailSender mailSender;
+
+    public RegistrationListener(UserService userService, MessageSource messages, JavaMailSender mailSender) {
+        this.userService = userService;
+        this.messages = messages;
+        this.mailSender = mailSender;
+    }
+
+    @Override
+    public void onApplicationEvent(OnRegistrationCompleteEvent event) {
+        this.confirmRegistration(event);
+    }
+
+    private void confirmRegistration(OnRegistrationCompleteEvent event) {
```

Comment:
Isn't a good name, still you creating  token, creating and sending email.

Reply:
Refactor this method